alicloud/resource_alicloud_polardb_cluster.go

```diff
@@ -0,0 +1,422 @@
+				Computed: true,
+			},
+			"effective_time": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{"Auto", "Immediately", "MaintainTime"}, false),
+				Optional:     true,
+				Default:      "Auto",
+			},
+		},
+	}
+}
+
+func resourceAlicloudPolarDBClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	polarDBService := PolarDBService{client}
+
+	request, err := buildPolarDBCreateRequest(d, meta)
+	if err != nil {
+		return WrapError(err)
+	}
+	raw, err := client.WithPolarDBClient(func(polarClient *polardb.Client) (interface{}, error) {
+		return polarClient.CreateDBCluster(request)
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_polardb_cluster", request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	response, _ := raw.(*polardb.CreateDBClusterResponse)
+	d.SetId(response.DBClusterId)
+
+	// wait cluster status change from Creating to running
+	stateConf := BuildStateConf([]string{"Creating"}, []string{"Running"}, d.Timeout(schema.TimeoutCreate), 5*time.Minute, polarDBService.PolarDBClusterStateRefreshFunc(d.Id(), []string{"Deleting"}))
+	if _, err := stateConf.WaitForState(); err != nil {
+		return WrapErrorf(err, IdMsg, d.Id())
+	}
+
+	return resourceAlicloudPolarDBClusterUpdate(d, meta)
+}
+
+func resourceAlicloudPolarDBClusterUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	polarDBService := PolarDBService{client}
+	d.Partial(true)
+
+	if d.HasChange("parameters") {
+		if err := polarDBService.ModifyParameters(d, d.Get("effective_time").(string), d.Get("parameters").(string)); err != nil {
+			return WrapError(err)
+		}
+		d.SetPartial("parameters")
+	}
+
+	if d.Get("cluster_charge_type").(string) == string(Prepaid) &&
+		(d.HasChange("renewal_status") || d.HasChange("auto_renew_period")) {
+		status := d.Get("renewal_status").(string)
+		request := polardb.CreateModifyAutoRenewAttributeRequest()
+		request.DBClusterIds = d.Id()
+		request.RenewalStatus = status
+
+		if status == string(RenewAutoRenewal) {
+			period := d.Get("auto_renew_period").(int)
+			request.Duration = strconv.Itoa(period)
+			request.PeriodUnit = string(Month)
+			if period > 9 {
+				request.Duration = strconv.Itoa(period / 12)
+				request.PeriodUnit = string(Year)
+			}
+		}
+
+		raw, err := client.WithPolarDBClient(func(polarDBClient *polardb.Client) (interface{}, error) {
+			return polarDBClient.ModifyAutoRenewAttribute(request)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+		// wait cluster status is Normal after modifying
+		stateConf := BuildStateConf([]string{"Creating"}, []string{"Running"}, d.Timeout(schema.TimeoutUpdate), 1*time.Minute, polarDBService.PolarDBClusterStateRefreshFunc(d.Id(), []string{"Deleting"}))
```

Comment:
Make sure the pending status is right, maybe it should be `Updating` or `Modifying`.